official/utils/arg_parsers/parsers.py

```diff
@@ -0,0 +1,208 @@
+          metavar="<MD>",
+      )
+
+
+class DeviceParser(argparse.ArgumentParser):
+  """Default parser for specifying compute devices (WIP).
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    multi_gpu: Create a flag to allow the use of all available GPUs.
+  """
+
+  def __init__(self, add_help=False, multi_gpu=False):
+    super(DeviceParser, self).__init__(add_help=add_help)
+
+    if multi_gpu:
+      self.add_argument(
+          "--multi_gpu", action="store_true",
+          help="If set, run across all available GPUs."
+      )
+
+
+class SupervisedParser(argparse.ArgumentParser):
+  """Default parser for specifying common supervised learning parameters.
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    train_epochs: Create a flag to specify the number of training epochs.
+    epochs_per_eval: Create a flag to specify the frequency of testing.
+    batch_size: Create a flag to specify the batch size.
+  """
+
+  def __init__(self, add_help=False, train_epochs=False, epochs_per_eval=False,
+               batch_size=False):
+    super(SupervisedParser, self).__init__(add_help=add_help)
+
+    if train_epochs:
+      self.add_argument(
+          "--train_epochs", "-te", type=int, default=1,
+          help="[default: %(default)s] The number of epochs used to train.",
+          metavar="<TE>"
+      )
+
+    if epochs_per_eval:
+      self.add_argument(
+          "--epochs_per_eval", "-epe", type=int, default=1,
+          help="[default: %(default)s] The number of training epochs to run "
+               "between evaluations.",
+          metavar="<EPE>"
+      )
+
+    if batch_size:
+      self.add_argument(
+          "--batch_size", "-bs", type=int, default=32,
+          help="[default: %(default)s] Batch size for training and evaluation.",
+          metavar="<BS>"
+      )
+
+
+class PerformanceParser(argparse.ArgumentParser):
+  """Default parser for specifying performance tuning arguments.
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    num_parallel_calls: Create a flag to specify parallelism of data loading.
+    inter_op: Create a flag to allow specification of inter op threads.
+    intra_op: Create a flag to allow specification of intra op threads.
+  """
+
+  def __init__(self, add_help=False, num_parallel_calls=False, inter_op=False,
+               intra_op=False):
+    super(PerformanceParser, self).__init__(add_help=add_help)
+
+    if num_parallel_calls:
+      self.add_argument(
+          "--num_parallel_calls", "-npc",
+          type=int, default=5,
+          help="[default: %(default)s] The number of records that are "
+               "processed in parallel  during input processing. This can be "
+               "optimized per data set but for generally homogeneous data sets,"
+               " should be approximately the number of available CPU cores.",
```

Comment:
Micro-nit: space should go at the end of the docstrings if that's where it is on the other lines.

Reply:
Fixed. There is a very sad one word line now...